src/components/Reports/SalaryCalculator/SalaryCalculatorContext/SalaryCalculatorContext.tsx

```diff
@@ -118,13 +125,22 @@ export const SalaryCalculatorProvider: React.FC<
   // End Step Handlers
 
   const [isDrawerOpen, setDrawerOpen] = useState(true);
+  const [hasStartedCalculation, setHasStartedCalculation] = useState(false);
   const { data: hcmData } = useHcmQuery();
   const { data: calculationData } = useSalaryCalculationQuery();
 
   const toggleDrawer = useCallback(() => {
     setDrawerOpen((prev) => !prev);
   }, []);
 
+  const startCalculation = useCallback(() => {
```

Comment:
This method should be moved to the landing page component since it's not used on the edit page. It should create a new salary calculation if the user doesn't have one yet.